Boot

Sindicat007 opened the pull request.

Sindicat007 committed:
Boot

Signed-off-by: Sindicat007 <sindicat007@ya.ru>

Sindicat007 committed:
Boot

Signed-off-by: Sindicat007 <sindicat007@ya.ru>

Sindicat007 committed:
Задача по созданию книжной библиотеки на Spring Boot

RinatBeybutov commented on SpringBoot/src/main/java/ru/maxima/spring/boot/config/SecurityConfig.java:
```diff
@@ -0,0 +1,55 @@
+public class SecurityConfig {
+
+    @Bean
+    UserDetailsService userDetailsService() {
```
public

RinatBeybutov commented on SpringBoot/src/main/java/ru/maxima/spring/boot/config/SecurityConfig.java:
```diff
@@ -0,0 +1,55 @@
+    public SecurityFilterChain securityFilterChain(HttpSecurity http) throws Exception {
+        return http
+                .authorizeHttpRequests(auth -> auth
+                        .requestMatchers("/login").permitAll()
```
permitAll означет доступ без авторизации. В задании написано, что books должен быть доступен пользователям после авторизации

RinatBeybutov commented on SpringBoot/src/main/java/ru/maxima/spring/boot/config/SecurityConfig.java:
```diff
@@ -0,0 +1,55 @@
+import ru.maxima.spring.boot.security.MyUserDetailsService;
+
+@Configuration
+@EnableWebSecurity
```
давай над каждым классом добавим javaDocs, как в приложении с чатом

RinatBeybutov commented on SpringBoot/src/main/java/ru/maxima/spring/boot/controller/BookController.java:
```diff
@@ -0,0 +1,60 @@
+    }
+
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
+    @PostMapping("/delete-book")
```
получится здесь delete mapping использовать? В задании указано, чтобы id был частью url, а не параметром запроса

Sindicat007 committed:
- Добавил JavaDoc
- Поменял маппинга на Delete в методе deleteBook
- Убрал общий доступ к списку книг
- Добавил модификатор доступа public для метода userDetailsService

Signed-off-by: Sindicat007 <sindicat007@ya.ru>

Sindicat007 committed:
Промежуточный коммит

Signed-off-by: Sindicat007 <sindicat007@ya.ru>

Sindicat007 committed:
- Поменял в методе deleteBook mapping

Signed-off-by: Sindicat007 <sindicat007@ya.ru>
